pandas/core/indexes/period.py

```diff
@@ -205,6 +210,18 @@ def _resolution_obj(self) -> Resolution:
         **_shared_doc_kwargs,
     )
     def asfreq(self, freq=None, how: str = "E") -> Self:
+        if isinstance(freq, str):
+            offset = to_offset(freq, is_period=True)
+            if hasattr(offset, "_period_dtype_code"):
+                freq = freq_to_period_freqstr(offset.n, offset.name)
+            elif offset.name == freq.replace(f"{offset.n}", ""):
+                raise ValueError(
+                    f"Invalid offset: '{offset.name}' for converting time series "
+                    f"with PeriodIndex."
+                )
```

Comment:
I don't really understand this, but I think you don't need it - just raise `ValueError(INVALID_FREQ_ERR_MSG.format(f"{freq}"))` in the `else` branch below?

Reply:
I use two different error messages here because we raise an error of invalid freq such as `“ABC”` and also for `freq=“”BMS”` which is valid for offsets, but invalid for period. I agree, that it can be confusing, so better to use the standard error message `"Invalid frequency: {freq}"`